esrally/track/loader.py

```diff
@@ -1128,8 +1128,9 @@ def __init__(self, track_params=None, complete_track_params=None, selected_chall
         self.source = source
         self.logger = logging.getLogger(__name__)
 
-    def __call__(self, track_name, track_specification, mapping_dir):
+    def __call__(self, track_name, track_specification, mapping_dir, spec_file=None):
         self.name = track_name
+        self.base_path = os.path.dirname(os.path.abspath(spec_file)) if spec_file else None
```

Comment:
should we add `self.base_path = None` in the constructor above? (like we do for `self.name`)

Reply:
yes, great point. I think this is a typical pylint violation that we're currently excluding actually